source/parser.hera

```diff
@@ -1323,7 +1323,7 @@ FunctionExpression
     }
 
   # Ruby/Crystal style block shorthand
-  AmpersandUnaryPrefix?:prefix Ampersand AmpersandBlockRHS?:rhs ->
+  AmpersandUnaryPrefix?:prefix ( Ampersand / ( !NumericLiteral &( QuestionMark? Dot ) ) ) AmpersandBlockRHS?:rhs ->
```

Comment:
What does the `!NumericLiteral` do?

Reply:
It prevents `.1` from being interpreted as `&.1`.

Reply:
Does it? Wouldn't it have to come after the Dot, inside the positive lookahead assertation?

Reply:
Oh, I see, I was thinking of `/[0-9]/`, but `NumericLiteral` includes a leading dot. Got it!